docs/designs/concepts.md

```diff
@@ -0,0 +1,164 @@
+# Concepts Design Doc
+
+This document describes core concepts used in ElasticDL, and discusses their representations.
+
+## Core Concepts
+
+Fundamentally, machine learning is using algorithms to extract information from raw data and represent it in some type of model.
+
+There are two kinds of parameters in a model, one is dense parameter, the other is sparse parameter. The sparse parameter is used by the embedding layer, so it is also called embedding table. The embedding table maps discrete ID *i* to embedding vector *vᵢ*.
+
+Both dense parameter and embedding table includes a tensor data field, and some other auxiliary fields, like name. Embedding table has an extra auxiliary field called indices.
+
+Tensor is used to represented a n-dim data. We support several data types, such as int8/int32/float32/float64.
+
+Besides, we have two kinds of gradients, dense gradient and embedding table gradient. The components of dense gradient is the same with dense parameter, and the components of embedding table gradient is also the same with embedding table.
```

Comment:
The components of dense gradient -> What's the meaning of component?

Reply:
Could fields or properties be more suitable?

Reply:
It means data structure. The data structure of dense gradient is the same with dense param.